pkg/patterns/declarative/metrics_test.go

```diff
@@ -529,26 +532,38 @@ func TestAddIfNotPresent(t *testing.T) {
 					t.Error(err)
 				}
 
+				tmpManifests, err := ioutil.TempFile("", "tmp-manifests-*.yaml")
+				if err != nil {
+					t.Error(err)
+				}
+				_, err = tmpManifests.WriteString(yobj)
+				if err != nil {
+					t.Error(err)
+				}
+				err = tmpManifests.Close()
+				if err != nil {
+					t.Error(err)
+				}
+				defer os.Remove(tmpManifests.Name())
```

Comment:
You can use `testing.T.TempDir()` to get a uniquely named test directory that gets cleaned up after the test finishes.
```suggestion
				tmpManifests, err := ioutil.TempFile(t.TempDir(), "tmp-manifests-*.yaml")
				if err != nil {
					t.Error(err)
				}
				_, err = tmpManifests.WriteString(yobj)
				if err != nil {
					tmpManifests.Close()
					t.Error(err)
				}
				err = tmpManifests.Close()
				if err != nil {
					t.Error(err)
				}
```